src/main/java/com/notably/logic/LogicManager.java

```diff
@@ -52,6 +55,15 @@ public void execute(String commandText) throws CommandException, ParseException
         }
     }
 
+    @Override
+    public void editCurrentBlockBody(String bodyContent) throws CannotModifyRootException {
```

Comment:
`CannotModifyRootException` is a `RuntimeException`, so you won't need to declare `throws` here!

If you want, you can create a custom checked exception in the `com.notably.logic.exceptions` package. We can call it something like `EditBlockBodyException`, for example!

We can then catch the `CannotModifyRootException` and rethrow an `EditBlockBodyException`!

Reply:
Ah i was thinking whether i should create a exception class for this sole purpose. Ok i'll do that